autoscheduler/scraper/management/commands/scrape_grades.py

```diff
@@ -0,0 +1,270 @@
+     """
+
+    loop = asyncio.get_running_loop()
+
+    async def retrieve_pdf(year_semester: str, college: str):
+        """ Creates a aiohttp session, downloads the grade report pdf, parses & scrapes it
+            and returns the list of scraped grades
+        """
+
+        async with ClientSession(loop=loop) as session:
+            pdf_path = await download_pdf(year_semester, college, session)
+
+            if pdf_path is not None:
+                # Parse the pdf then scrape the returned grade distributions
+                print(f"Parsing PDF for {year_semester} {college}")
+
+                grade_dists = pdf_parser.parse_pdf(pdf_path)
+
+                # Convert the year_semester term to include the location
+                term = generate_term_with_location(year_semester, college)
+
+                scraped_grades = scrape_pdf(grade_dists, term)
+
+                return scraped_grades
+
+            return None
+
+    semesters = [SPRING, SUMMER, FALL]
+
+    # Gather all of the retrieve functions into an array
+    tasks = []
+    for year in years:
+        for semester in semesters:
+            for college in colleges:
+                year_semester_term = str(year) + semester
+                tasks.append(retrieve_pdf(year_semester_term, college))
+
+    grades = [] # list of Grades models
+    for scraped_grades in await asyncio.gather(*tasks, loop=loop):
+        if scraped_grades:
+            grades.extend(scraped_grades)
+
+    return grades
+
+def fetch_page_data() -> bs4.BeautifulSoup:
+    """ Retrieves the ROOT_URL HTML data and converts it to BeautifulSoup
+        so we can parse it to get the available years & colleges to scrape grades for
+    """
+
+    response = requests.get(ROOT_URL, verify=True)
+    response.raise_for_status()
+
+    soup = bs4.BeautifulSoup(response.text, "lxml")
+
+    return soup
+
+class Command(base.BaseCommand):
+    """ Retrieves all of the possible colleges & terms from web.as-tamu.edu
+        Then downloads all of the respective PDF's and parses them
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('--year', '-y', type=int,
+                            help="The year you want to scrape grades for, such as 2019")
+        parser.add_argument('--college', '-c', type=str,
+                            help="The college you want to scrape grades for, such as EN")
+
+    def handle(self, *args, **options):
+        start = time.time()
+
+        # Create the documents/grade_dists folder if it doesn't already exist
+        _create_documents_folder()
+
+        # Retrieve the page data and get the available colleges & years from it
+        page_soup = fetch_page_data()
+        years = _get_available_years(page_soup)
+        if options['year']:
+            years = [options['year']]
+
+        colleges = _get_colleges(page_soup)
+        if options['college']:
+            colleges = [options['college']]

```

Comment:
Change to a ternary or if/else so we don't have to call `select` unless the values are being used